src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -1868,16 +1868,69 @@ private static Expression BindParameterFromValue(ParameterInfo parameter, Expres
+
+            var splitAndTrimMethod = typeof(RequestDelegateFactory).GetMethod(nameof(SplitAndTrim), System.Reflection.BindingFlags.Static | System.Reflection.BindingFlags.NonPublic);
+            if (splitAndTrimMethod == null)
+            {
+                throw new InvalidOperationException("The method 'SplitAndTrim' could not be found on 'RequestDelegateFactory'.");
+            }
+            var splitAndTrimExpr = Expression.Call(splitAndTrimMethod, headerValuesArrayExpr);
+
+            var boundValueExpr = Expression.Convert(splitAndTrimExpr, parameter.ParameterType);
+
+            if (!isOptional)
+            {
+                var checkRequiredStringParameterBlock = Expression.Block(
+                    Expression.Assign(argument, boundValueExpr),
+                    Expression.IfThen(Expression.Equal(argument, Expression.Constant(null)),
+                        Expression.Block(
+                            Expression.Assign(WasParamCheckFailureExpr, Expression.Constant(true)),
+                            Expression.Call(LogRequiredParameterNotProvidedMethod,
+                                HttpContextExpr, parameterTypeNameConstant, parameterNameConstant, sourceConstant,
+                                Expression.Constant(factoryContext.ThrowOnBadRequest))
+                        )
+                    )
+                );
+
+                // NOTE: when StringValues is used as a parameter, value["some_unpresent_parameter"] returns StringValue.Empty, and it's equivalent to (string?)null
+                factoryContext.ExtraLocals.Add(argument);
+                factoryContext.ParamCheckExpressions.Add(checkRequiredStringParameterBlock);
+                return argument;
+            }
+
+            // Allow nullable parameters that don't have a default value
+            if (nullability.ReadState != NullabilityState.NotNull && !parameter.HasDefaultValue)
+            {
+                if (parameter.ParameterType == typeof(StringValues?))
+                {
+                    return Expression.Block(
+                        Expression.Condition(Expression.Equal(boundValueExpr, Expression.Convert(Expression.Constant(StringValues.Empty), parameter.ParameterType)),
+                            Expression.Convert(Expression.Constant(null), parameter.ParameterType),
+                            boundValueExpr
+                        )
+                    );
+                }
+                return boundValueExpr;
+            }
+
+            return Expression.Block(
+                Expression.Condition(Expression.NotEqual(boundValueExpr, Expression.Constant(null)),
+                    boundValueExpr,
+                    Expression.Convert(Expression.Constant(parameter.DefaultValue), parameter.ParameterType)));
+        }
+
         if (!isOptional)
         {
-            // The following is produced if the parameter is required:
-            //
-            // argument = value["param1"];
```

Comment:
Please re-add (and update) this. It's very useful to have a comment showing what the generated code is doing.